flask_server.py

```diff
@@ -71,27 +71,42 @@ def index():
     cookie_path = settings["cookie_path"] if "cookie_path" in settings else "token.txt"
     folder_path = settings["folder_path"]  if "folder_path" in settings else "ig_media"
     if request.method == "POST":
-        amountScraped = int(request.form["amountToScrape"])
+        if not os.path.exists("token.txt"):
+            log_lines = get_log_file_list() 
+            return render_template('index.html', count_i= count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = False, logged_in = False)
+        if request.form["amountToScrape"].isnumeric():
+            amountScraped = int(request.form["amountToScrape"])
+        else:
+            log_lines = get_log_file_list() 
+            return render_template('index.html', count_i= count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = True, logged_in = True)
         mode = request.form["mode_dropdown"]
         base64_media = start_scrape(cookie_path, folder_path, amountScraped, mode)
         rendered_base64_media = render_base64_media(base64_media)
     log_lines = get_log_file_list()   
     if len(log_lines) > 0:
         count_u, count_i, count_v = get_stats_from_log_line(log_lines)      
-    return render_template('index.html', count_i = count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media)
+    return render_template('index.html', count_i = count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = False, logged_in = True)
 
 @app.route("/settings/", methods=['GET','POST'])
 def settings():
+    credentials_error = False
     settings = get_settings()    # Gets the settings
     if request.method == "POST":
         updated_settings = settings
         for setting in request.form:
             if len(request.form[setting]) > 0:
                 updated_settings[setting] = request.form[setting]
         save_settings(updated_settings)
+        if request.form["username"] and request.form["password"] != "":
```

Comment:
No need to check != "" if X and Y is enough.
BTW X and Y != K just checks Y !=K, you have to do X != K and Y != K in similar cases